### ISSUE TITLE
Slow completions when editing Python code

### ISSUE THREAD

**Issue by DonJayamanne** [microsoft/vscode#210528]
Reported by @ale-dg here https://github.com/microsoft/vscode/issues/206119#issuecomment-1996165888

Yes, but consider I did not use Pylance. Would you like me to test with it? I just activated it and opened the log trace for python and just for indexing it finishes its memory.

2024-03-13 17:51:01.092 [info] [Info - 17:51:01] (4192) Heap stats: total_heap_size=1220MB, used_heap_size=1156MB, total_physical_size=1218MB, total_available_size=2900MB, heap_size_limit=4096MB

2024-03-13 17:51:01.108 [info] [Warn - 17:51:01] (4192) Workspace indexing has hit its upper limit: 2000 files

**Comment by ale-dg** [comment]
Alright then. Thanks for the help! Hope it also helps in solving the other issue regarding the speed of execution. 

Best

**Comment by DonJayamanne** [comment]
Please can you provide a sample csv file used in your notebook

**Comment by ale-dg** [comment]
Sure, attached the file.

Best

[palmerpenguins_extended.csv](https://github.com/microsoft/vscode/files/15014999/palmerpenguins_extended.csv)

**Comment by heejaechang** [comment]
don't notebook vs regular hover/signaturehelp use different markdown renderer?

**Comment by DonJayamanne** [comment]
> don't notebook vs regular hover/signaturehelp use different markdown renderer?

From what I can tell they are the same,

**Comment by mjbvz** [comment]
Yes the parameter hints widget is the same everywhere, it's only notebook output that uses a different markdown renderer

**Comment by DonJayamanne** [comment]
@mjbvz 

Sometimes I have seen this taking 300ms when working on notebooks, we use markdown parser to build outlines and that ends up parsing markdown which results in 300ms sometimes.
So, this is definitely happening in other places too, but not 100% of the time.
The issue does definitely repro for me (again not 100% of the time), please let me know if CPU profiles (however we have this from the user) will help and what else information you need.

**Comment by leifwalsh** [comment]
These are long issues so forgive me if this isn't new information, but I'm pretty sure I have the same problem described here and I've noticed something that might help?

When I type code into a cell and then try to execute it, when I get this laggy behavior, I notice inlay error messages like this show up:

![image](https://github.com/microsoft/vscode/assets/46406/1a70f030-3914-43dc-b30e-8f20a3b995ff)

As I watch (not touching anything, just waiting for the cell to execute), it progressively changes, as if it's analyzing prefixes of the content in the cell, or as if I were typing each character into the cell one by one very slowly. Here's another one (it made progress while I was typing):

![image](https://github.com/microsoft/vscode/assets/46406/ac3e71a2-5b6b-48f7-b773-ab36ca67968b)

I also, in this notebook where I just reproduced it, had a perfectly fine experience when I started from an empty notebook, for about 13 cells worth of mixed code and markdown, then around the 14th cell something changed and it started doing the bad thing. It was very sudden, like I'd crossed a threshold or something, which surprised me because before I've only seen it sometimes, when opening notebooks I'd already authored (and this reminded me to try reading these issues again).

I'd be happy to share a reproducer, gather diagnostics, etc. but it looks like you have many? If you want, just point me to a message with instructions for gathering the ones that would help right now, I see you've done that quite a lot already. :)

**Comment by DonJayamanne** [comment]
Thank you, 
We have made a lot of performance improvements in vscode for editing and executing notebooks
Please can you install the latest vscode insiders and the latest pre-release version of Jupyter extension and test this out

let me know how it goes

If you still run into issues please share the cpu logs  while you are experiencing these same issues
Instructions here https://github.com/microsoft/vscode/issues/210528#issuecomment-2060334576

**Comment by carlos-andrade8** [comment]
Hi Guys,

I came across to this issue regarding the slowness of jupiter notebooks. I'm experiencing performance issues (extremely slow) when I try to print the shape of a dataframe or to show the head of the dataframe. It only happens when using:

pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)
pd.set_option('display. width', 1000)

It tries to render but it takes so long. I'm not sure if the above is related to this issue.

Thanks,
